mbed_tools_ci/assert_news.py

```diff
@@ -20,136 +18,86 @@
+            raise ValueError(f'Empty news file "{self._basename}".')
         if len(file_content.splitlines()) > 1:
-            raise ValueError(f'News file "{basename}" contains more than one line.')
+            raise ValueError(
+                f'News file "{self._basename}" contains more than one line.'
+            )
 
     def validate(self) -> None:
         """Verifies news file follows standards."""
-        logger.info(f'Verifying {self._news_file_path}')
+        logger.info(f"Verifying {self._basename}")
         self.validate_file_name()
         self.validate_file_contents()
 
 
-class NewsFileDiscoverer:
-    """Checks that all new PRs comprise a news file and that such files follow the standard."""
-
-    def __init__(self) -> None:
-        """Creates an instance of NewsFileDiscoverer.
-
-        Set up the git wrapper and save references to the current and master branches
-        """
-        self.git = GitWrapper()
-        self.current_branch = self.git.get_current_branch()
-        self.master_branch = self.git.get_master_branch()
-
-    def find_news_file(self) -> List[str]:
-        """Determines a list of all the news files which were added as part of the PR.
-
-        Returns:
-             list of introduced news files
-        """
-        news_dir = configuration.get_value(ConfigurationVariable.NEWS_DIR)
-        if not os.path.exists(news_dir):
-            NotADirectoryError(
-                f'News files directory was not specified and default path `{news_dir}` does not exist'
-            )
-        logger.info(
-            f':: Looking for news files in `{news_dir}` [{self.current_branch}]'
-        )
-        self.git.checkout(self.current_branch)
-        self.git.set_upstream_branch(self.current_branch)
-        self.git.pull()
-        current_commit = self.git.get_current_commit()
-        # Delete `master` as it may already exist and be set to the one of interest (on CircleCI anyway and maybe on other systems)
-        if self.git.branch_exists(self.master_branch):
-            self.git.delete_branch(self.master_branch)
-        self.git.checkout(self.master_branch)
-        self.git.set_upstream_branch(self.master_branch)
-        self.git.pull()
-        master_branch_commit = self.git.get_current_commit()
-        self.git.checkout(self.current_branch)
-        project_root = configuration.get_value(
-            ConfigurationVariable.PROJECT_ROOT)
-        news_dir_relative_path = os.path.relpath(news_dir, start=project_root)
-        added_news = self.git.get_changes_list(
-            self.git.get_branch_point(
-                master_branch_commit, current_commit),
-            current_commit, change_type='a',
-            dir=news_dir_relative_path
-        )
-        extension_to_exclude = ['.toml', '.rst']
-        return [path for path in added_news if
-                len([ex for ex in extension_to_exclude if ex in path]) == 0]
+def validate_news_file(absolute_path: str) -> None:
```

Comment:
I much prefer not using classes if we don't have to 👍 I find it difficult to test code where I have to create full instances of classes just to test small internal methods.